actions/trickle-down-changelog/index.js

```diff
@@ -0,0 +1,119 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import * as semver from 'semver';
+import * as exec from "@actions/exec"
+import * as fs from "fs/promises";
+
+const octokit = github.getOctokit(core.getInput("token"));
+const simpleSemverRegex = /\d+\.\d+\.\d+(-.*)?/;
+const internalBotEmail = "internal@elementor.com";
+async function main() {
+	const currentRef = github.context.ref.replace("refs/heads/", "");
+
+	// we only care about merges to beta/ga branches
+	if (!semver.parse(currentRef) && !semver.parse(currentRef + ".0"))
+		return;
+
+	const commitInfo = await octokit.request("GET /repos/{owner}/{repo}/commits/{sha}", {
+		owner: github.context.repo.owner,
+		repo: github.context.repo.repo,
+		sha: github.context.sha,
+	});
+
+	// if pr opened with the internal bot, no need to continue
+	if (commitInfo.data.commit.author.email === internalBotEmail)
+		return;
+
+	const diff = await octokit.request("GET /repos/{owner}/{repo}/commits/{sha}", {
+		owner: github.context.repo.owner,
+		repo: github.context.repo.repo,
+		sha: github.context.sha,
+		headers: {
+			accept: "application/vnd.github.diff"
+		}
+	});
+	const changedVersions = getVersions(diff);
+	const oldest = getOldestVersionFromChanged(changedVersions);
+
+	const branches = await octokit.request("GET /repos/{owner}/{repo}/branches", {
+		owner: github.context.repo.owner,
+		repo: github.context.repo.repo,
+	});
+
+	const gitBranches = branches.data.filter((branch) => {
+		const toSemver = semver.parse(branch.name + ".0");
+		if (toSemver)
+			return semver.gt(toSemver.version, oldest);
+	});
+	const branchesToPRTo = gitBranches.map((branch) => branch.name);
+
+	// always need to pr to main
+	branchesToPRTo.push("main");
+
+	console.log(`branches to pr: ${branchesToPRTo}`)
+	const changelog = await fs.readFile("changelog.txt");
+	let readmeContent = undefined;
+	if (github.context.repo.repo === "elementor") {
```

Comment:
We usually use yoda conditions
```suggestion
	if ("elementor" === github.context.repo.repo) {
```

Reply:
hate yoda conditions, is it alright if we leave it to the php?